code/game/objects/items/drug_items.dm

```diff
@@ -0,0 +1,70 @@
+/obj/item/food/drug
+	name = "generic drug"
+	desc =  "I am error"
+	icon = 'icons/obj/drugs.dmi'
+	foodtypes = GROSS
+	food_flags = FOOD_FINGER_FOOD
+	max_volume = 50
+	eat_time = 1 SECONDS
+	tastes = list("drugs" = 2, "chemicals" = 1)
+	eatverbs = list("gnaws" = 1)
+	bite_consumption = 10
+	w_class = WEIGHT_CLASS_TINY
+
+/obj/item/food/drug/saturnx
+	name = "saturnX glob"
+	desc = "A congealed glob of pure saturnX.\nThis compound was first discovered during the infancy of cloaking technology and at the time thought to be a promising candidate agent. It was withdrawn for consideration after the researchers discovered a slew of associated safety issues including thought disorders and hepatoxicity.\nIt has since attained some limited popularity as a street drug."
+	icon_state = "saturnx_glob" //tell kryson to sprite two more variants in the future.
+	food_reagents = list(/datum/reagent/drug/saturnx = 10)
+
+/obj/item/food/drug/moon_rock
+	name = "moon rock"
+	desc = "A small hard lump of kronkaine freebase.\nIt is said the average kronkaine addict causes as much criminal damage as four cat burglars, two arsonists and one rabid pit bull terrier combined."
+	icon_state = "moon_rock1"
+	food_reagents = list(/datum/reagent/drug/kronkaine = 10)
+
+/obj/item/food/drug/moon_rock/Initialize()
+	. = ..()
+	icon_state = pick("moon_rock1", "moon_rock2", "moon_rock3")
+
+/obj/item/reagent_containers/glass/blastoff_ampoule
+	name = "bLaSToFF ampoule" //stylized name
+	desc = "A small ampoule. The liquid inside appears to be boiling violently.\nYou suspect it contains bLasSToFF; the drug thought to be the cause of the infamous Luna nightclub mass casualty incident."
+	icon = 'icons/obj/drugs.dmi'
+	icon_state = "blastoff_ampoule"
+	base_icon_state = "blastoff_ampoule"
+	volume = 20
+	reagent_flags = TRANSPARENT
+	spillable = FALSE
+	list_reagents = list(/datum/reagent/drug/blastoff = 10)
+
+/obj/item/reagent_containers/glass/blastoff_ampoule/update_icon_state()
+	. = ..()
+	if(!reagents.total_volume)
+		icon_state = "[base_icon_state]_empty"
+	else if(spillable)
+		icon_state = "[base_icon_state]_open"
+	else
+		icon_state = base_icon_state
+
+/obj/item/reagent_containers/glass/blastoff_ampoule/attack_self(mob/user)
+	if(user.canUseTopic(src, BE_CLOSE, NO_DEXTERITY) && !spillable)
+		reagent_flags |= OPENCONTAINER
+		spillable = TRUE
+		playsound(src, 'sound/items/ampoule_snap.ogg', 40)
+		update_appearance()
+		return
+	. = ..()
+
+/obj/item/reagent_containers/glass/blastoff_ampoule/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
+	if(!.) //if the bottle wasn't caught
```

Comment:
you don't set . to anything, this is by default null and thus will always trigger. also, you need to early return this.
```dm
. = ..()
if(.)
    return
rest of your code here
```